File: e-nutrition-symfony/src/Entity/Mesure.php
<?php

namespace App\Entity;

use App\Repository\MesureRepository;
use Doctrine\ORM\Mapping as ORM;

/**
 * @ORM\Entity(repositoryClass=MesureRepository::class)
 */
class Mesure
{
    /**
     * @ORM\Id
     * @ORM\GeneratedValue
     * @ORM\Column(type="integer")
     */
    private $id;

    /**
     * @ORM\Column(type="float")
     */
    private $Taille;

    /**
     * @ORM\Column(type="float")
     */
    private $poids;

    /**
     * @ORM\Column(type="datetime")
     */
    private $dateMesure;

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTaille(): ?float
    {
        return $this->Taille;
    }

    public function setTaille(float $Taille): self
    {
        $this->Taille = $Taille;

        return $this;
    }

    public function getPoids(): ?float
    {
        return $this->poids;
    }

    public function setPoids(float $poids): self
    {
        $this->poids = $poids;

        return $this;
    }

    public function getDateMesure(): ?\DateTimeInterface
    {
        return $this->dateMesure;
    }

    public function setDateMesure(\DateTimeInterface $dateMesure): self
    {
        $this->dateMesure = $dateMesure;

        return $this;
    }

    public function calculerIMC(){
        return round(($this->poids / ($this->Taille ** 2)), 2);
    }

    public function poidsIdeal(){
        // 50 + (0.91 × [height in centimeters − 152.4])
        return 50 + (0.91 * ($this->Taille * 100 - 152.4));
    }
}

File: e-nutrition-mobile/src/e/nutrition/entities/Mesure.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package e.nutrition.entities;

import java.util.Date;

/**
 *
 * @author Ayoub
 */
public class Mesure {
    
    private Integer id;
    private float taille;
    private float poids;
    private Date date_mesure;
    private Patient patient_id;

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public float getTaille() {
        return taille;
    }

    public void setTaille(float taille) {
        this.taille = taille;
    }

    public float getPoids() {
        return poids;
    }

    public void setPoids(float poids) {
        this.poids = poids;
    }

    public Date getDate_mesure() {
        return date_mesure;
    }

    public void setDate_mesure(Date date_mesure) {
        this.date_mesure = date_mesure;
    }

       public Patient getPatient_id() {
        return patient_id;
    }

    public void setPatient_id(Patient patient_id) {
        this.patient_id = patient_id;
    }
    
    public Mesure() {
    }

    public Mesure(float taille, float poids, Date date_mesure , Patient patient_id) {
        this.taille = taille;
        this.poids = poids;
        this.date_mesure = date_mesure;
        this.patient_id = patient_id;
    }

    @Override
    public String toString() {
        return "Mesure{" + "id=" + id + ", taille=" + taille + ", poids=" + poids + ", date_mesure=" + date_mesure + ", user_patient=" + patient_id + '}';
    }

    public void InitSelectedBlog(Mesure mesure) {
        throw new UnsupportedOperationException("Not supported yet."); //To change body of generated methods, choose Tools | Templates.
    }

  
    
    
}
